drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,801 @@
+*******************************************************************************/
+int32_t adxl355_get_xyz(struct adxl355_dev *dev,
+			float *x,
+			float *y,
+			float *z)
+{
+	uint8_t array_raw_x[3];
+	uint8_t array_raw_y[3];
+	uint8_t array_raw_z[3];
+	int32_t status = 0;
+
+	status |= adxl355_get_n_consec_reg_val(dev,
+					       ADXL355_XDATA3,
+					       ADXL355_XDATA_BYTE_SIZE,
+					       array_raw_x);
+	*x = adxl355_accel_convertion(dev, array_raw_x);
+
+	status |= adxl355_get_n_consec_reg_val(dev,
+					       ADXL355_YDATA3,
+					       ADXL355_YDATA_BYTE_SIZE,
+					       array_raw_y);
+	*y = adxl355_accel_convertion(dev, array_raw_y);
+
+	status |= adxl355_get_n_consec_reg_val(dev,
+					       ADXL355_ZDATA3,
+					       ADXL355_ZDATA_BYTE_SIZE,
+					       array_raw_z);
+	*z = adxl355_accel_convertion(dev, array_raw_z);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the raw temperature data and converts it to degrees Celsius.
+ *
+ * @param dev      - The device structure.
+ * @param temp     - Temperature output data.
+ *
+ * @return status  - Result of the reading procedure.
+*******************************************************************************/
+int32_t adxl355_get_temp(struct adxl355_dev *dev,
+			 float *temp)
+{
+	uint8_t array_raw_temp[2];
+	int32_t status = 0;
+
+	status |= adxl355_get_n_consec_reg_val(dev,
+					       ADXL355_TEMP2,
+					       ADXL355_ZDATA_BYTE_SIZE,
+					       array_raw_temp);
+	*temp = adxl355_temp_convertion(array_raw_temp);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the status register value.
+ *
+ * @param dev          - The device structure.
+ * @param status_flags - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int32_t adxl355_get_status_register(struct adxl355_dev *dev,
+				    struct adxl355_status_register_flags *status_flags)
+{
+	int32_t status;
+	uint8_t reg_value;
+	status = adxl355_get_register_value(dev, ADXL355_STATUS, &reg_value);
+
+	if (status == 0) {
+		status_flags->NVM_BUSY  = (reg_value >> 4) & 1;
+		status_flags->Activity  = (reg_value >> 3) & 1;
+		status_flags->FIFO_OVR  = (reg_value >> 2) & 1;
+		status_flags->FIFO_FULL = (reg_value >> 1) & 1;
+		status_flags->DATA_RDY  = reg_value & 1;
+	}
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the number of FIFO entries register value.
+ *
+ * @param dev       - The device structure.
+ * @param reg_value - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int32_t adxl355_get_nb_of_fifo_entries(struct adxl355_dev *dev,
```

Comment:
is this function needed? `adxl355_get_register_value(dev, ADXL355_FIFO_ENTRIES, reg_value);` can be used inline where needed.